packages/framework/src/Model/Pricing/PriceConverter.php

```diff
@@ -34,26 +35,45 @@ public function __construct(CurrencyFacade $currencyFacade, Rounding $rounding)
     /**
      * @param \Shopsys\FrameworkBundle\Component\Money\Money $price
      * @param int $domainId
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency|null $priceCurrency
      * @return \Shopsys\FrameworkBundle\Component\Money\Money
      */
-    public function convertPriceWithoutVatToPriceInDomainDefaultCurrency(Money $price, int $domainId): Money
+    public function convertPriceWithoutVatToPriceInDomainDefaultCurrency(Money $price, int $domainId, ?Currency $priceCurrency = null): Money
     {
-        $currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
-        $price = $price->divide($currency->getExchangeRate(), static::DEFAULT_SCALE);
+        $domainDefaultCurrency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
+        $price = $this->convertPriceToPriceInDomainDefaultCurrency($price, $domainDefaultCurrency, $priceCurrency);
 
         return $this->rounding->roundPriceWithoutVat($price);
     }
 
     /**
      * @param \Shopsys\FrameworkBundle\Component\Money\Money $price
      * @param int $domainId
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency|null $priceCurrency
      * @return \Shopsys\FrameworkBundle\Component\Money\Money
      */
-    public function convertPriceWithVatToPriceInDomainDefaultCurrency(Money $price, int $domainId): Money
+    public function convertPriceWithVatToPriceInDomainDefaultCurrency(Money $price, int $domainId, ?Currency $priceCurrency = null): Money
     {
-        $currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
-        $price = $price->divide($currency->getExchangeRate(), static::DEFAULT_SCALE);
+        $domainDefaultCurrency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
+        $price = $this->convertPriceToPriceInDomainDefaultCurrency($price, $domainDefaultCurrency, $priceCurrency);
 
-        return $this->rounding->roundPriceWithVatByCurrency($price, $currency);
+        return $this->rounding->roundPriceWithVatByCurrency($price, $domainDefaultCurrency);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $price
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency $domainDefaultCurrency
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency|null $priceCurrency
+     * @return \Shopsys\FrameworkBundle\Component\Money\Money
+     */
+    protected function convertPriceToPriceInDomainDefaultCurrency(Money $price, Currency $domainDefaultCurrency, ?Currency $priceCurrency = null): Money
+    {
+        if ($priceCurrency === null) {
+            trigger_error('$priceCurrency argument must be provided', E_USER_DEPRECATED);
+            $priceCurrency = $this->currencyFacade->getByCode(Currency::CODE_CZK);
```

Comment:
Can I ask why CZK? 
How it will behave when I don't set $priceCurrency? Is it possible some tests will fail?

Reply:
because right now, we are always sending prices in CZK currency to `PriceConverter`, see data fixtures and tests. So I wanted to keep this setting when no `$priceCurrency` is provided. I would not say that any test would fail if you do not set `$priceCurrency`